src/server/services/registrar.js

```diff
@@ -164,8 +167,85 @@ async function getShortname(offeredTermCode, classSectionID) {
   }
 }
 
+/**
+ * Gets the week number, given an academic term and a date
+ *
+ * @param {string} term Academic term, formatted as YYQ (e.g. 20S)
+ * @param {string} date Date string formatted as MM/DD/YYYY
+ * @returns {?string} Returns week number if date is valid in term
+ */
+async function getWeekNumber(term, date) {
+  try {
+    // If Week Number for date is already cached, return that
+    if (cache.has(`WeekNumber_${date}`)) {
```

Comment:
@chern This is an unnecessary cache to me.. how often will dates overlap across multiple medias?